tests/warn_on_potential_data_race_test.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+import warnings
+import dace
+import numpy as np
+import pytest
+
```

Comment:
These tests are OK, but you may want to add more tests for data races. Examples:

1.
```
(A) -> (B)
(C) -> (A)
```

2.
```
(A) -> map -> (B)
(A) -> map -> (B)
```
(i.e., with two different access nodes)

3. symbolic overlap (2*i, i)
4. constant memlet overlap (3:10, 6:12) and "almost-overlaps" (3:10, 10:20)